ui/src/app/shared/components/revision.tsx

```diff
@@ -1,9 +1,13 @@
 import * as React from 'react';
 import {revisionUrl} from './urls';
 
-export const Revision = ({repoUrl, revision, children}: {repoUrl: string; revision: string; children?: React.ReactNode}) => {
+export const Revision = ({repoUrl, revision, path, children}: {repoUrl: string; revision: string; path?: string; children?: React.ReactNode}) => {
     revision = revision || '';
-    const url = revisionUrl(repoUrl, revision);
+    const hasPath = path && path !== '.';
+    let url = revisionUrl(repoUrl, revision, hasPath);
+    if (hasPath) {
+        url += '/' + path;
```

Comment:
This will still create a URL for unsupported sources, where `url` is `null`, e.g. 
<img width="497" alt="Screenshot 2023-08-02 at 4 16 17 PM" src="https://github.com/argoproj/argo-cd/assets/928827/81da2bc3-7a0f-4ece-98c2-817b98ad4115">